flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/FileArchivedExecutionGraphStoreTest.java

```diff
@@ -0,0 +1,319 @@
+	@Test
+	public void testUnknownGet() throws IOException {
+		final File rootDir = temporaryFolder.newFolder();
+
+		try (final FileArchivedExecutionGraphStore executionGraphStore = createDefaultExecutionGraphStore(rootDir)) {
+			assertThat(executionGraphStore.get(new JobID()), Matchers.nullValue());
+		}
+	}
+
+	/**
+	 * Tests that we obtain the correct jobs overview.
+	 */
+	@Test
+	public void testStoredJobsOverview() throws IOException {
+		final int numberExecutionGraphs = 10;
+		final Collection<ArchivedExecutionGraph> executionGraphs = generateTerminalExecutionGraphs(numberExecutionGraphs);
+
+		final List<JobStatus> jobStatuses = executionGraphs.stream().map(ArchivedExecutionGraph::getState).collect(Collectors.toList());
+
+		final JobsOverview expectedJobsOverview = JobsOverview.create(jobStatuses);
+
+		final File rootDir = temporaryFolder.newFolder();
+
+		try (final FileArchivedExecutionGraphStore executionGraphStore = createDefaultExecutionGraphStore(rootDir)) {
+			for (ArchivedExecutionGraph executionGraph : executionGraphs) {
+				executionGraphStore.put(executionGraph);
+			}
+
+			assertThat(executionGraphStore.getStoredJobsOverview(), Matchers.equalTo(expectedJobsOverview));
+		}
+	}
+
+	/**
+	 * Tests that we obtain the correct collection of available job details.
+	 */
+	@Test
+	public void testAvailableJobDetails() throws IOException {
+		final int numberExecutionGraphs = 10;
+		final Collection<ArchivedExecutionGraph> executionGraphs = generateTerminalExecutionGraphs(numberExecutionGraphs);
+
+		final Collection<JobDetails> jobDetails = executionGraphs.stream().map(WebMonitorUtils::createDetailsForJob).collect(Collectors.toList());
+
+		final File rootDir = temporaryFolder.newFolder();
+
+		try (final FileArchivedExecutionGraphStore executionGraphStore = createDefaultExecutionGraphStore(rootDir)) {
+			for (ArchivedExecutionGraph executionGraph : executionGraphs) {
+				executionGraphStore.put(executionGraph);
+			}
+
+			assertThat(executionGraphStore.getAvailableJobDetails(), Matchers.containsInAnyOrder(jobDetails.toArray()));
+		}
+	}
+
+	/**
+	 * Tests that an expired execution graph is removed from the execution graph store.
+	 */
+	@Test
+	public void testExecutionGraphExpiration() throws Exception {
+		final File rootDir = temporaryFolder.newFolder();
+
+		final Time expirationTime = Time.milliseconds(1L);
+
+		final ManuallyTriggeredScheduledExecutor scheduledExecutor = new ManuallyTriggeredScheduledExecutor();
+
+		try (final FileArchivedExecutionGraphStore executionGraphStore = new FileArchivedExecutionGraphStore(
+			rootDir,
+			expirationTime,
+			10000L,
+			scheduledExecutor)) {
+
+			final ArchivedExecutionGraph executionGraph = new ArchivedExecutionGraphBuilder().setState(JobStatus.FINISHED).build();
+
+			executionGraphStore.put(executionGraph);
+
+			// there should one execution graph
+			assertThat(executionGraphStore.size(), Matchers.equalTo(1));
+
+			Thread.sleep(expirationTime.toMilliseconds());
```

Comment:
Isn't it possible to use a `Ticker` to test the time based eviction? https://github.com/google/guava/wiki/CachesExplained#testing-timed-eviction

Reply:
Will change it.